ivy/data_classes/container/base.py

```diff
@@ -2728,7 +2728,7 @@ def cont_set_at_key_chains(self, target_dict, return_dict=None, inplace=False):
                 return_dict[k] = self.cont_set_at_key_chains(v, return_dict[k], inplace)
             else:
                 return_dict[k] = v
-        return ivy.Container(return_dict, **self._config)
+        return return_dict if inplace else ivy.Container(return_dict, **self._config)
```

Comment:
I don't think this line makes a difference in the behaviour of the `inplace` argument, as the `return_dict` is always a `Container`. 

I'm not sure why we need the `ivy.Container(return_dict, **self.config)` in the first place, as we're retaining the config even when we do at line [`2725`](https://github.com/unifyai/ivy/pull/18164/files#diff-d15a6b35d36e632e8493df8bdfbcf9a027cb4599dccac4f41ca077788892fdc2R2725), and that's the only possible use of calling the `ivy.Container.__init__` with `return_dict` being a container (i.e. if the `config` of `return_dict` wasn't correct after calling `cont_copy` but that isn't the case). 
@saeedashrraf any ideas?

In the meantime, could you please revert this change @RickSanchezStoic 😅?

Reply:
It does make a difference because even for `inplace=True` a new container is returned with the **config added. Earlier, I too just added a `return return_dict` but since wasn't sure about the config thing so I omitted it. This indeed is a bug since it doesn't update the code inplace if `inplace=True`. Let me know if you have more doubts @vedpatwardhan

Reply:
That's what we are doing right? We need to pass the values then to `self.v`  like `self.v=created_n_found` , so better do this in one line. @vedpatwardhan

Reply:
I think just to ensure that the function always returns an `ivy.Container`, we should maybe change [2731](https://github.com/unifyai/ivy/pull/18164/files#diff-d15a6b35d36e632e8493df8bdfbcf9a027cb4599dccac4f41ca077788892fdc2R2731)  to be `return return_dict if inplace and isinstance(return_dict, ivy.Container) else ivy.Container(return_dict, **self._config)` . 

Let me know what you think

Reply:
Actually, no need to check for `inplace`, we can just change it to be `return return_dict if isinstance(return_dict, ivy.Container) else ivy.Container(return_dict, **self._config)`

Reply:
> Actually, no need to check for `inplace`, we can just change it to be `return return_dict if isinstance(return_dict, ivy.Container) else ivy.Container(return_dict, **self._config)`

This sounds good, what do you say @vedpatwardhan

Reply:
> Actually, no need to check for `inplace`, we can just change it to be `return return_dict if isinstance(return_dict, ivy.Container) else ivy.Container(return_dict, **self._config)`

Yeah this sounds perfect, let's update `set_at_key_chains` and other relevant instance methods with this handling @RickSanchezStoic, thanks @saeedashrraf 🙂